#!/usr/bin/env python
PACKAGE = "roomac_autonomous_manipulation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("object_position_correction_x", double_t, 0, "Correction offset applied to object position", 0.0, -0.5, 0.5)
gen.add("object_position_correction_y", double_t, 0, "Correction offset applied to object position", 0.0, -0.5, 0.5)
gen.add("object_position_correction_z", double_t, 0, "Correction offset applied to object position", 0.0, -0.5, 0.5)

gen.add("pre_goal_retraction_x", double_t, 0, "Offset applied to object position as a pre gripping position", 0.1, 0, 0.5)
gen.add("post_goal_retraction_z", double_t, 0, "Offset applied to object position as a post gripping position", 0.08, 0, 0.5)

gen.add("open_gripper_wait_time", double_t, 0, "How long to wait for gripper to open", 0.0, 0, 4)
gen.add("close_gripper_wait_time", double_t, 0, "How long to wait for gripper to close", 1.0, 0, 4)

gen.add("procedure_retry_threshold", int_t, 0, "Number of retries of each procedure", 5, 0, 10)

exit(gen.generate(PACKAGE, "roomac_autonomous_manipulation", "PickManagerSettings"))